tensorflow/lite/micro/tools/requantize_flatbuffer_utils.py

```diff
@@ -0,0 +1,215 @@
+  return clip_range(vals, bit_width)
+
+
+def dequantize_data(quantized_data, scale, zero_point=0):
+  """Dequantize the data to integer type with desired bit width.
+
+  Args:
+      quantized_data (np.array): quantized data
+      scale (float): quantization scale of the data
+      zero_point (integer): quantization zero point of the data
+
+  Returns:
+      np.array : dequantized data
+  """
+  return scale * (quantized_data - zero_point)
+
+
+'''Conversion Util Functions'''
+
+
+def change_quantization_settings_8to16(tensor):
+  if not tensor.quantization:
+    return
+  assert (tensor.quantization.quantizedDimension == 0
+          )  # Only layer quantization supported
+  scale, zero_point = (
+      tensor.quantization.scale[0],
+      tensor.quantization.zeroPoint[0],
+  )
+  # Asymmertical quantized: scale * (qmax - zero_point) = rmax
+  rmax = scale * (MAX_INT8 - zero_point)
+  rmin = scale * (MIN_INT8 - zero_point)
+  # symmertical quantized: scale * qmax = rmax
+  scale_16 = max(abs(rmax), abs(rmin)) / abs(MIN_INT16)
+  # Change scale: Symmetrical Quantized
+  tensor.quantization.scale = [scale_16]
+  tensor.quantization.zeroPoint = [0]
+
+
+def change_activation_tensor_8to16(tensor):
+  if tensor.type == TENSOR_TYPE_CODE[np.int8]:
+    # change quantization settings
+    change_quantization_settings_8to16(tensor)
+    # Set tensor type
+    tensor.type = TENSOR_TYPE_CODE[np.int16]
+    logging.info(f"Set {tensor.name} from int8 to int16 ")
+
+
+def set_bias_type_int64(buffers, input, weight, bias):
+  bias_buffer = buffers[bias.buffer]
+  bias_scale, bias_zero_pt = (
+      bias.quantization.scale[0],
+      bias.quantization.zeroPoint[0],
+  )
+  if len(bias_buffer.data):
+    data = np.frombuffer(bias_buffer.data, dtype=np.int32)
+    dequantized_data = dequantize_data(data, bias_scale, bias_zero_pt)
+    bias_scale_int64 = (input.quantization.scale[0] *
+                        weight.quantization.scale[0])
+    bias_zero_pt_int64 = 0  # symmetrical quantized
+    int64_data = quantize_data(dequantized_data, bias_scale_int64,
+                               bias_zero_pt_int64, 64).astype(np.int64)
+    bias_buffer.data = int64_data.tobytes()
+
+  # Set tensor type
+  bias.type = TENSOR_TYPE_CODE[np.int64]
+  bias.quantization.scale = [bias_scale_int64]
+  bias.quantization.zeroPoint = [bias_zero_pt_int64]
+  logging.info(f"Set {bias.name} from int32 to int64")
+  '''Specific op conversion functions'''
+
+
+def requantize_fully_connected(tensors, buffers, op):
+  input_tensor = tensors[op.inputs[0]]
+  weight_tensor = tensors[op.inputs[1]]
+  bias_tensor = None
+  if op.inputs[2] >= 0:
+    bias_tensor = tensors[op.inputs[2]]
+  output_tensor = tensors[op.outputs[0]]
+
+  # Change Activation type
+  change_activation_tensor_8to16(input_tensor)
+  change_activation_tensor_8to16(output_tensor)
+  # Change Bias
+  if bias_tensor:
+    set_bias_type_int64(buffers, input_tensor, weight_tensor, bias_tensor)
```

Comment:
since nias is not being used about, might be more readable to combine the logic.

```py
if len(op.inputs) >= 2:
  bias_tensor = tensorspop.inputs[2]]
  set_bias_type_int64(buffers, input_tensor, weight_tensor, bias_tensor)
```

Reply:
Yes it is. Changed with the condition  op.inputs[2]!= -1